src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,182 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int currentSize;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (currentSize == 0) {
+            addFirstNode(value);
+            return;
+        }
+        addNodeToTailPosition(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > currentSize) {
+            throw new IndexOutOfBoundsException("Input index : " + index
+                    + " out of bonds: [" + 0 + ", " + currentSize + "]");
```

Comment:
redundant string concat
```suggestion
                    + " out of bonds: [0 - "  + currentSize + "]");
```